pkg/agent/bgp/engine/logs.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package engine
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	gobgplog "github.com/osrg/gobgp/v3/pkg/log"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/natefinch/lumberjack.v2"
+
+	"antrea.io/antrea/pkg/util/logdir"
+)
+
+const (
+	logfileSubdir string = "bgp"
+	logfileName   string = "gobgp.log"
+)
+
+// goBGPServerLogger implements github.com/osrg/gobgp/v3/pkg/log/Logger interface.
+type goBGPServerLogger struct {
+	logger      *logrus.Logger
+	extraFields map[string]interface{}
+}
```

Comment:
Is there a strong requirement to have a separate log file for bgp? Why we don't implement the required loger using klog to get unified logs?

Reply:
Not a strong requirement. I thought it would be more convenient for users to debug BGP if the corresponding logs are located in a separate log file, like `/var/log/antrea/networkpolicy` for NetworkPolicy, `/var/log/antrea/ovs` for OVS.

> Why we don't implement the required loger using klog to get unified logs?

That's fine with me. I'll try to implement it.

Reply:
`/var/log/antrea/networkpolicy` is for traffic logs, not process logs. I think `/var/log/antrea/ovs` is more "have to" as they are individual processes. In my mind gobpg log is not very different from memberlist and ofnet logs.